experimental/embeddings/datasets.py

```diff
@@ -0,0 +1,132 @@
+
+
+def split_train_test_dev(data: List[Any]) -> Dict[str, List[Any]]:
+    train, test = train_test_split(data, test_size=0.4, random_state=1)
+    dev, test = train_test_split(test, test_size=0.5, random_state=1)
+    return {"train": train, "dev": dev, "test": test}
+
+
+class Dataset:
+    @abc.abstractmethod
+    def to_flair_column_corpus(self) -> ColumnCorpus:
+        pass
+
+
+def get_dataset_cls(
+    name: str, datasets_module: str = "experimental.embeddings.datasets"
+) -> Any:
+    try:
+        return getattr(importlib.import_module(datasets_module), name)
+    except AttributeError:
+        raise NotImplementedError(f"Dataset {name} not supported!")
+
+
+class PromisesElectionsPLDataset(Dataset):
+    def __init__(self, url: Optional[str] = None):
+        super().__init__()
+        self.ds_files = ["train.csv", "dev.csv", "test.csv"]
+        self.url = url
+        self.path = "resources/datasets/promises-elections-pl/"
+        if not self._detect_files(path=self.path, files=self.ds_files):
+            if url:
+                print("Downloading and preprocessing data.")
+                self._preprocess()
+            else:
+                raise ValueError("Can't download files without download url being set.")
+
+    @staticmethod
+    def _detect_files(path: str, files: List[str]) -> bool:
+        return all([os.path.exists(os.path.join(path, file)) for file in files])
+
+    def _download(self) -> str:
+        assert isinstance(self.url, str)
+        downloader = DatasetDownloader(
+            root_dir="resources/datasets/promises-elections-pl/",
+            url=self.url,
+        )
+        downloaded = downloader.download()
+        if not isinstance(downloaded, str):
+            raise ValueError("Download failed. Expected one file.")
+
+        return downloaded
+
+    def _preprocess(self) -> None:
+        path = self._download()
+        data = list(srsly.read_jsonl(path))
+        splitted_data = split_train_test_dev(data)
+        for key, ds in splitted_data.items():
+            convert_jsonl_to_connl(ds, out_path=os.path.join(self.path, f"{key}.csv"))
+        os.remove(path)
+
+    def to_flair_column_corpus(self) -> ColumnCorpus:
+        return ColumnCorpus(
+            data_folder=self.path,
+            column_format={0: "text", 1: "tag"},
+            train_file="train.csv",
+            test_file="test.csv",
+            dev_file="dev.csv",
+        )
+
+
+class DatasetDownloader:
+    def __init__(self, root_dir: str, url: str, filename: Optional[str] = None):
+        self.root_dir = root_dir
+        self.url = url
+        self.filename = filename
+
+        self._chunk_size = 1024
+
+    @property
+    def _dl_path(self) -> str:
+        assert isinstance(self.filename, str)
+        return os.path.join(self.root_dir, self.filename)
+
+    def _download_file(self) -> None:
+        r = requests.get(self.url, stream=True)
+        assert r.status_code == 200
```

Comment:
asserts are used for debugging on the developing level, they can be disabled ( https://docs.python.org/3/using/cmdline.html#envvar-PYTHONOPTIMIZE ). imo it should by replaced by raising some exception.

Reply:
On this example I agree that It should be replaced for exceptions. But in the context of other parts of code asserts were used only for mypy reasons. As I have small experience with this lib, suggestions for ensuring types when using mypy are welcome.